extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/CdcSinks.java

```diff
@@ -0,0 +1,335 @@
+            @Nonnull FunctionEx<ChangeRecord, K> keyFn,
+            @Nonnull FunctionEx<ChangeRecord, V> valueFn
+    ) {
+        String name = "remoteMapCdcSink(" + map + ')';
+        return sink(name, map, clientConfig, keyFn, valueFn);
+    }
+
+    @Nonnull
+    private static <K, V> Sink<ChangeRecord> sink(
+            @Nonnull String name,
+            @Nonnull String map,
+            @Nullable ClientConfig clientConfig,
+            @Nonnull FunctionEx<ChangeRecord, K> keyFn,
+            @Nonnull FunctionEx<ChangeRecord, V> valueFn
+    ) {
+        ProcessorSupplier supplier = AbstractHazelcastConnectorSupplier.of(asXmlString(clientConfig),
+                instance -> new CdcSinkProcessor<>(instance, map, keyFn, extend(valueFn)));
+        ProcessorMetaSupplier metaSupplier = ProcessorMetaSupplier.forceTotalParallelismOne(supplier, name);
+        return new SinkImpl<>(name, metaSupplier, true, null);
+    }
+
+    @Nonnull
+    private static <V> FunctionEx<ChangeRecord, V> extend(@Nonnull FunctionEx<ChangeRecord, V> valueFn) {
+        return (record) -> {
+            if (DELETE.equals(record.operation())) {
+                return null;
+            }
+            return valueFn.apply(record);
+        };
+    }
+
+    private static class CdcSinkProcessor<K, V> extends UpdateMapWithMaterializedValuesP<ChangeRecord, K, V> {
+
+        private Sequences<K> sequences;
+
+        CdcSinkProcessor(
+                @Nonnull HazelcastInstance instance,
+                @Nonnull String map,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends K> keyFn,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends V> valueFn
+        ) {
+            super(instance, map, keyFn, valueFn);
+        }
+
+        @Override
+        public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+            super.init(outbox, context);
+
+            HazelcastProperties properties = new HazelcastProperties(context.jetInstance().getConfig().getProperties());
+            long expirationMs = properties.getMillis(SEQUENCE_CACHE_EXPIRATION_SECONDS);
+            this.sequences = new Sequences<>(expirationMs);
+        }
+
+        @Override
+        protected boolean shouldBeDropped(K key, ChangeRecord item) {
+            ChangeRecordImpl recordImpl = (ChangeRecordImpl) item;
+            long sequencePartition = recordImpl.getSequencePartition();
+            long sequenceValue = recordImpl.getSequenceValue();
+
+            boolean isNew = sequences.update(key, sequencePartition, sequenceValue);
+            return !isNew;
+        }
+
+    }
+
+    /**
+     * Tracks the last seen sequence number for a set of keys. The
+     * sequence numbers originate from Debezium event headers and consist
+     * of two parts.
+     * <p>
+     * The <i>sequence</i> part is exactly what the name implies: a numeric
+     * sequence which we base our ordering on. Implementations needs to ensure
```

Comment:
this is not true, it's a specific three element array and the role of each element is well defined. it's just not any array that's sorted

Reply:
The comment talks about the conceptual sequence number, which consists of two parts, not the array we use, which also holds a timestamp used for expiration, but which is not part of the sequence number. Will find a clearer form.

Reply:
Yes that's kind of what I meant.. the description didn't quite match what was in the code (with the long[])